### PR TITLE
consistent deployments names

### PR DESCRIPTION
**Pull request by Omarabdul3ziz**
### Changes
- Fixing the panic in `destroy` method due to missing client connection.
- Explicitly set the types for functions that takes `solutionType` as argument.
- Define type for keys in `solutionsList`.
- Edit the `deployer` function to set the `solutionType` as `projectName` in grid-client.
- Unify the types name to be Uppercase and required to be type of `IStore["type"]`

### Related Issues
- https://github.com/threefoldtech/grid_weblets/issues/1289
- https://github.com/threefoldtech/grid_weblets/issues/1336

### Checklist
- [x] Test passed Deploying/Listing/Deleting almost all weblets.
- [x] Build pass

### CODE REVIEW

**Comment by xmonader** [comment]
how would that be backward compatible?

**Comment by Omarabdul3ziz** [comment]
Added a backward compatible listing. Now the deployment can be listed from 3 sources.

1. default namespace *no project name provided while deploying* it is filtered by the solution name in the flist. (uppercase/lowercase)
2. namespace when the project name provided in lowercase.
3. namespace when the project name provided in uppercase *the implemented way in this pr*


This is applied also on Kubernetes deployments with project names `""`, `"k8s"`, and `"Kubernetes"`

**Comment by 0oM4R** [comment]
as I discussed with Rabie; the problem was We shouldn't import file from weblets in vue dashboard, so the solution is to create a new type for the istore interface, and import it in vue dashboard, also to avoid conflicts as the istore type used in many places, we export it at [src/stores/currentDeployment.ts](https://github.com/threefoldtech/grid_weblets/pull/1337/files#diff-1696e245fad1ce9e4cd609c7e45220078613529b365ab74c3a38213e92b9e9ad)

File: src/elements/DeployedList/DeployedList.wc.svelte [hunk 1]
```diff
@@ -43,33 +18,34 @@
...
-    { label: "Taiga", value: "taiga" },
-    { label: "Owncloud", value: "owncloud" },
-    { label: "Presearch", value: "presearch" },
-    { label: "Subsquid", value: "subsquid" },
-    { label: "Casperlabs", value: "casperlabs" },
-    { label: "Algorand", value: "algorand" },
-    //{ label: "TFhub Validator", value: "tfhubValidator" },
-    { label: "Node Pilot", value: "nodepilot" },
-    { label: "QSFS Virtual Machine", value: "qvm"},
-    { label: "Umbrel", value: "umbrel"},
```
Comment (AhmedHanafy725) [inline src/elements/DeployedList/DeployedList.wc.svelte:67]: why umbrel is removed?

File: src/stores/solutionsList.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-  presearch: "ps",
-  caprover: "cp",
-  kubernetes: "k8s",
-  machines: "vm",
-  nodepilot: "np",
-  fullvm: "fvm",
-  subsquid: "ss",
-  algorand: "al",
-  qvm: "qvm",
-  umbrel: "um",
```
Comment (0oM4R) [inline src/stores/solutionsList.ts:20]: umbrel removed here as well

**Comment by Mahmoud-Emad** [comment]
The code looks good to me, but i can't test it to ensure that the changes don't reflect anything else because the grid proxy is down

**Review by Mahmoud-Emad — APPROVED** [review]
The code looks good to me, but i can't test it to ensure that the changes don't reflect anything else because the grid proxy is down

**Comment by 0oM4R** [comment]
I tried to deploy `wordpress` and `umberl` and `subsquid` on play.dev.grid.tf
and list, and delete them using this pr
also to verify the related issue #1289 i manually threw an error before deploying gw in `wordpress`, the deployment rolled back successfully
image


anyway, I guess it's important to mention that deleting solutions that have gw not deleting gw contract,
Screenshot from 2023-03-21 13-20-46
I tried it on `subsquid` and `wordpress` and it happens on both of them, this happens on play.dev.grid.tf so I guess this issue is unrelated to this pr

**Pull request merged**